### ISSUE TITLE
Support for the "event" messages over JSON-RPC.

### ISSUE THREAD

**Issue by piotrpartyka** [joncol/jcon-cpp#17]
Hi!
Is there a way to support the "event" type messages by this library? After receiving an event I'd like to send a signal with the content of the message, catch it in my own class and then parse it. For now I added some code in JsonRpcClient::jsonResponseReceived for letting the "event" messages go, but after that when I try to send some request over call() method the library gets stuck inside my added code and then the app crashes. 

Please help me :)
Piotr

This is the code I added. The `receivedEvent()` signal is added to the class.
```
void JsonRpcClient::jsonResponseReceived(const QJsonObject response)
{
    //qDebug() << "Message received:" << response;
    //JCON_ASSERT(response["jsonrpc"].toString() == "2.0");

    if(response.contains(QString("event"))){
        qDebug() << "Event: " << response;
        emit receivedEvent(response);
        return;
    }

    if (response.value("jsonrpc").toString() != "2.0") {
      . . .
```

**Comment by piotrpartyka** [comment]
@joncol, could you please jhelp me? :smiley:

**Comment by joncol** [comment]
Hello! Could you clarify some more what you want to do? Are you writing both client and server? It sounds like you need to create what I refer to as a `Service` (see the readme), which will contain your event handler. Your client code should then call some (RPC) method of this service with suitable arguments.

It sounds like you shouldn't need to dig around inside the library code. Or am I missing something?

**Comment by piotrpartyka** [comment]
Hi! Thanks for the response. I'll clarify my case.
I've got a server, which is a music player - it's called Mopidy. I try talk with it using your library. I already had to make some changes, because the server understands some requests without the "params:" key and the library always adds it to the request. 

The case was to enable the library to accept the messages form the server without the "jsonrpc:" key and with "event:" key. That's the way the server works. 
Fortunately I got it figured out myself and now everything is working great :smiley: 

I'd like to thank you for making this library. It helps me a lot in my project and to be honest I don't know what I would do without it. Probably I would have to write it myself, but it would take me a lot of time and effort, and the result wouldn't be even close to your lib. So thanks again, great job! :medal_sports:
